galpy/potential/TwoPowerSphericalPotential.py

```diff
@@ -181,8 +315,143 @@ def _dens(self,R,z,phi=0.,t=0.):
         HISTORY:
            2010-08-08 - Written - Bovy (NYU)
         """
-        r= numpy.sqrt(R**2.+z**2.)
-        return (self.a/r)**self.alpha/(1.+r/self.a)**(self.beta-self.alpha)/4./m.pi/self.a**3.
+        a, alpha, beta = self.a, self._alpha, self._beta
+        r= sqrt(R**2.+z**2.)
+        return (a/r)**alpha/(1.+r/a)**(beta-alpha)/4./m.pi/a**3.
+
+    def _mass(self,R,z=0.,t=0.):
+        """
+        NAME:
+           _mass
+        PURPOSE:
+           evaluate the mass within R for this potential
+        INPUT:
+           R - Galactocentric cylindrical radius
+           z - vertical height
+           t - time
+        OUTPUT:
+           the mass enclosed
+        HISTORY:
+           2014-04-01 - Written - Erkal (IoA)
+        """
+        a, alpha, beta = self.a, self._alpha, self._beta
+        r= R if z is None else sqrt(R**2.+z**2.)
+        # if z is None:
+        #     r= R
+        # else:
+        #     r= sqrt(R**2.+z**2.)
+        return (r/a)**(3.-alpha)/(3.-alpha)*hyp2f1(3.-alpha,-alpha+beta,4.-alpha,-r/a)
+
+
+class DehnenSphericalPotential(TwoPowerSphericalPotential):
+    """Class that implements the Dehnen Spherical Potential from Dehnen 1993
+
+    .. math::
+
+          \\rho(r) = \\frac{\\mathrm{amp}(3-\\alpha)}{4\\,\\pi\\,a^3}\\,\\frac{1}{(r/a)^{\\alpha}\\,(1+r/a)^{4-\\alpha}}
+    """
+    def __init__(self,amp=1.,a=1.,alpha=1.5,dehnen_amp=False,
+                 normalize=False,ro=None,vo=None):
+        """
+        NAME:
+
+           __init__
+
+        PURPOSE:
+
+           initialize a Dehnen Spherical Potential
+
+        INPUT:
+
+           amp - amplitude to be applied to the potential (default: 1); can be a Quantity with units of mass or Gxmass will be rescaled to amp*(3-alpha) (except for Jaffe, which is already scaled)
+
+           a - scale radius (can be Quantity)
+
+           alpha - inner power, restricted to [0, 3)
+
+           dehnen_amp - if True, use dehnen normalization, amp -> amp * (3 - alpha)
```

Comment:
Not sure we should support this, it's pretty trivial, but just adds a use-case to test...

Reply:
I put in in there so that the default `DehnenSphericalPotential` would match `HernquistPotential` when initialized with the same arguments. switching `dehnen_amp` to True makes it match the Dehnen 1993 paper. This is also useful if `alpha` is ever permitted to vary.

Reply:
I would prefer just keeping a single amp definition and just putting a note in the docs that this definition differs from the Dehnen paper. It will just be less confusing and require less testing of different cases...

Reply:
Ok. I've made dehnen_amp=False the norm. This will show up in the next push.

Reply:
I think you should remove the ``dehnen_amp`` option altogether, not just change the default.

Reply:
removed.